src/snowflake/snowpark/_internal/compiler/large_query_breakdown.py

```diff
@@ -124,6 +126,10 @@ def __init__(
         self.complexity_score_upper_bound = (
             session.large_query_breakdown_complexity_bounds[1]
         )
+        # This is used to track the number of partitions we could not breakdown because we
+        # could not find any valid nodes. We also track if we could have broken down the plan
+        # only if externally referenced CTEs were considered valid.
+        self.breakdown_failure_summary = defaultdict(int)
```

Comment:
instead of add it as a field, maybe do it similar as how we do it for the CTE reference, just return a structure result RepeatedSubqueryEliminationResult with the result summary. Or update the other one to do the same thing, so that we have a uniformed way of handling the optimizations

Reply:
done